src/software/thunderscope/field/obstacle_layer.py

```diff
@@ -0,0 +1,49 @@
+import pyqtgraph as pg
+from pyqtgraph.Qt import QtCore, QtGui
+from software.thunderscope.field.field_layer import FieldLayer
+import software.thunderscope.field.constants as constants
+import software.thunderscope.colors as colors
+from proto.geometry_pb2 import Polygon, Circle
+from proto.visualization_pb2 import Obstacle
+from software.networking.threaded_unix_listener import ThreadedUnixListener
+
+
+class ObstacleLayer(FieldLayer):
+    def __init__(self):
+        FieldLayer.__init__(self)
+        self.obstacle_receiver = ThreadedUnixListener(
+            "/tmp/tbots/TbotsProto.Obstacle", Obstacle, max_buffer_size=1,
+        )
+        self.cached_obstacles = Obstacle()
+
+    def paint(self, painter, option, widget):
+
+        obstacles = self.obstacle_receiver.maybe_pop()
+
+        if not obstacles:
+            obstacles = self.cached_obstacles
+
+        self.cached_obstacles = obstacles
+
+        painter.setPen(pg.mkPen(colors.NAVIGATOR_OBSTACLE_COLOR))
+
+        for polyobstacle in obstacles.polygon:
+            polygon_points = [
+                QtCore.QPoint(
+                    constants.MM_TO_M * point.x_meters,
+                    constants.MM_TO_M * point.y_meters,
```

Comment:
I'm pretty sure this is converting m to mm, but the name of the constant MM_TO_M is a bit confusing since it sounds like it's converting from mm to m, could we rename it to M_TO_MM or MM_PER_M